munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala

```diff
@@ -49,23 +49,29 @@ trait ScalaCheckSuite extends FunSuite {
 
   private def propToTry(prop: Prop, test: Test): Try[Unit] = {
     import ScalaCheckTest._
-    def seed =
+    def makeSeed() =
       scalaCheckTestParameters.initialSeed.getOrElse(
         Seed.fromBase64(scalaCheckInitialSeed).get
       )
+    val initialSeed = makeSeed()
+    var seed: Seed = initialSeed
     val result = check(
       scalaCheckTestParameters,
-      Prop(genParams => prop(genParams.withInitialSeed(seed)))
+      Prop { genParams =>
+        val r = prop(genParams.withInitialSeed(seed))
+        seed = seed.next
+        r
+      }
     )
     def renderResult(r: Result): String = {
       val resultMessage = Pretty.pretty(r, scalaCheckPrettyParameters)
       if (r.passed) {
         resultMessage
       } else {
-        val seedMessage = s"""|Failing seed: ${seed.toBase64}
+        val seedMessage = s"""|Failing seed: ${initialSeed.toBase64}
```

Comment:
If a test failed, say 67th out of 100, we still report the 1st seed in case the test is not pure.